### PR TITLE
Fix nil error in OCP

### PR DESCRIPTION
**Pull request by hhovsepy**
Kiali pod is failing with nil error on OCP cluster in `.isK8sGatewayListerInit`.

Seems checking only cache lister `c.getCacheLister(namespace).k8sgatewayLister == nil` is causing `nil` in case of cluster or namespace scope of caches. 
Added a `nil` check similar to `GetK8sGateways` method structure.

### CODE REVIEW

**Comment by hhovsepy** [comment]
Kiali run fine now on OCP.

File: kubernetes/cache/kube_cache.go [hunk 1]
```diff
@@ -1467,12 +1467,27 @@ func (c *kubeCache) GetTelemetries(namespace, labelSelector string) ([]*v1alpha1
 }
 
 func (c *kubeCache) isK8sGatewayListerInit(namespace string) bool {
-	// K8s GW has several cache listers, those can be namespace or cluster based
-	// if one of them is not initialized, then all others are not init as well
-	// this can happen when CRDs are created after Kiali start
-	if c.getCacheLister(namespace).k8sgatewayLister == nil {
```
Comment (nrfox) [inline kubernetes/cache/kube_cache.go:1473]: Looks like the issue is that `c.getCacheLister(namespace)` can return `nil`? Would probably be simpler to just add a conditional here then that checks for that before accessing `k8sgatewayLister`. Something like:
```
if cacheLister := c.getCacheLister(namespace); cacheLister == nil || cacheLister.k8sgatewayLister == nil {
  return false
}
```

**Review by nrfox — CHANGES_REQUESTED** [review]

File: kubernetes/cache/kube_cache.go (hunk 1 above)
Comment (hhovsepy) [reply kubernetes/cache/kube_cache.go:1473]: then it will always return false on OCP cluster and the Gateway functionality could be missed

Comment (hhovsepy) [reply kubernetes/cache/kube_cache.go:1473]: yeah agree, I had that idea on my previous PR, but avoided adding a new variable, now it sound better idea than many checks

**Comment by hhovsepy** [comment]
@nrfox ready for a review

**Review by nrfox — APPROVED** [review]

**Review by jshaughn — COMMENTED** [review]
Sorry for late review, but this approach looks good.

**Pull request merged**